lookout/style/format/feature_extractor.py

```diff
@@ -728,21 +717,30 @@ def _parse_file(self, contents: str, root: bblfsh.Node, path: str) -> \
             pos = node.end_position.offset
         return result, parents
 
-    def _remove_labels_with_low_support(self, vnodes: Sequence[VirtualNode]) -> None:
+    def _compute_labels_mappings(self, vnodes: Sequence[VirtualNode]) -> None:
         """
-        Calculate the support for each label and discard those with too little value.
+        Calculate the label to class sequence and class sequence to label mappings.
+
+        Takes into account self.cutoff_label_support and discard those with too little value.
 
         :param vnodes: The virtual nodes extracted from all the files.
         """
-        label_support = defaultdict(int)
+        assert len(self.class_sequences_to_labels) == 0, self.class_sequences_to_labels
+        assert len(self.labels_to_class_sequences) == 0, self.labels_to_class_sequences
+        support = defaultdict(int)
         for vnode in vnodes:
             if vnode.y is not None:
-                label_support[vnode.y] += 1
-        unsupported = {k for k, v in label_support.items() if v < self.cutoff_label_support}
-        for vnode in vnodes:
-            if vnode.y in unsupported:
-                vnode.y = None
-        self._log.debug("Removed %d/%d labels by support %d", len(unsupported), len(label_support),
+                support[vnode.y] += 1
+
+        for class_seq in sorted(support, key=lambda l: -support[l]):
```

Comment:
O(n log n) instead of O(n) and less readable

Reply:
if it was not an optimization but a feature it's fine though

Reply:
It is small array ~100 of items, so does not really affect performance.
I just think it is a good idea to have this order: from the most frequent label to the least frequent.

Reply:
yup I agree

Reply:
The for loop does nothing